### DISCUSSION TITLE
Is it possible to login with RemoteCommand defined

### DISCUSSION THREAD

**Discussion by kemeris2000** [pyinfra-dev/pyinfra#1130] [General]
I connect to all provisioned VM's via jumphost which forward connections to VM's vsocket based on SSH Remote Command like this:

`ssh root@jumphost 127`

in this example remote command "127" is VM ID and jumphost forward connection to corresponding vsocket VM is listening to.

Is it possible to set SSH RemoteCommand parameter for login in pyinfra?

**Comment by xvello** [comment]
Hello @kemeris2000, looking at the doc this is not currently supported, but since `pyinfra` allows [passing a ssh_paramiko_connect_kwargs argument](https://docs.pyinfra.com/en/3.x/connectors/ssh.html), you might be able to achieve this in the Paramiko side. I'd check their documentation and issues for similar use cases.

Possibly a dud, but you might be able to achieve this by specifying [a custom transport_factory](https://docs.paramiko.org/en/latest/api/client.html) that wraps [the default Transport implem](https://docs.paramiko.org/en/latest/api/transport.html) and executes your desired connect command before returning the session to the caller.

**Reply by kemeris2000** [reply]
Thank you for response @xvello, will look to paramiko documentation. I was trying to define `RemoteCommand` in custom ssh_config

ssh_config_custom.conf
```
Host *
  RemoteCommand 112

```
and pass this file to inventory host like this:

hosts = [
   ("host1",
      {"ssh_user": "user",
       "ssh_password": "password",
       "ssh_key": "",
       "ssh_config_file": "/var/lib/pyinfra/ssh_config_custom.conf"}
   ),
]

but unfortunately it does not work, -vvv output does show sufficient info to debug. With standard linux ssh client, ssh_config_custom.conf does work

**Reply by xvello** [reply]
I can confirm that passing a `ssh_config_file` path in the inventory data works to set a specific username or port. The `Paramiko` library does not use the `openssh` client but re-implements the protocol itself. This means that not all options supported by `openssh` will be supported by `Paramiko`. Unfortunately, [the documentation](https://docs.paramiko.org/en/latest/api/config.html?highlight=SSHConfig#config-module-api-documentation) does not list what options are implemented. There's no code search match for `RemoteCommand`, so it''s safe to assume they don't implement it.

Could be worth contributing the feature upstream directly? The end of [SSHClient.connect](https://github.com/paramiko/paramiko/blob/51eb55debf2ebfe56f38378005439a029a48225f/paramiko/client.py#L217) might be the right place to add this.

**Reply by kemeris2000** [reply]
Thank you for your help @xvello

**Reply by xvello** [reply]
Heya @kemeris2000, other people could have the same use case and appreciate it if you could share how you ended up implementing it.

**Reply by kemeris2000** [reply]
Unfortunately I didn't find solution, so I am forced to use ansible.
